clippy_lints/src/write.rs

```diff
@@ -266,9 +302,13 @@ impl EarlyLintPass for Write {
 /// ```
 /// will return
 /// ```rust,ignore
-/// (Some("string to write: {}"), Some(buf), false)
+/// (Some(("string to write: {}"), StrStyle::Cooked, "1:15-1:36"), Some(buf))
 /// ```
-fn check_tts<'a>(cx: &EarlyContext<'a>, tts: &TokenStream, is_write: bool) -> (Option<String>, Option<Expr>, bool) {
+fn check_tts<'a>(
+    cx: &EarlyContext<'a>,
+    tts: &TokenStream,
+    is_write: bool,
+) -> (Option<(String, StrStyle, Span)>, Option<Expr>, bool) {
```

Comment:
oof. Maybe it's the Time to define a struct for this return type and document it properly 😄

Reply:
Factored the tuple into its own struct and removed the redundant bool parameter.